baseclasses/problems/pyWeight_problem.py

```diff
@@ -69,9 +68,9 @@ def addComponents(self, components):  # *components?
         """
 
         # Check if components is of type Component or list, otherwise raise Error
-        if type(components) == list:
+        if isinstance(components,list):
             pass
-        elif type(components) == object:
+        elif isinstance(components,object):
```

Comment:
Everything in Python is an object so idk if this does anything

Reply:
Yeah I'll leave that decision up to you guys, but the quality of code in this module (and elsewhere in this repo) is quite poor.